VAULT-12226: Add Static Roles to the AWS plugin (#20536)
hashicorp/vault: website/content/api-docs/secret/aws.mdx

  }
}
```

## Create Static Role
This endpoint creates or updates static role definitions. A static role is a 1-to-1 mapping
with an AWS IAM User, which will be adopted and managed by vault, including rotating it according
to the configured `rotation_period`.

> -> Vault will create a new credential upon configuration, and if the maximum number of access keys are already
> in existence, Vault will rotate the oldest one. Vault must do this in order to know the credential.
> 
> -> Each rotation, Vault will rotate the oldest existing credential.

Reviewer: ```suggestion
<Note>

Vault will create a new credential upon configuration, and if the maximum number of access keys already exist, Vault will rotate the oldest one. Vault must do this to know the credential. 

At each rotation, Vault will rotate the oldest existing credential.

</Note>
```

  }
}
```

> ## Create Static Role

Reviewer 2: Not sure if this is already planned, but it'd be neat to have a mention on the [docs](https://developer.hashicorp.com/vault/docs/secrets/aws) page that this feature exists! Might be hard to discover in API docs alone.

Author: I threw something very basic up just as a pointer.

Author: (pinging @yhyakuna in case she wants to take a look)

This endpoint creates or updates static role definitions. A static role is a 1-to-1 mapping
with an AWS IAM User, which will be adopted and managed by Vault, including rotating it according
to the configured `rotation_period`.

<Note>

Vault will create a new credential upon configuration, and if the maximum number of access keys already exist, Vault will rotate the oldest one. Vault must do this to know the credential. 

At each rotation, Vault will rotate the oldest existing credential.

</Note>

| Method | Path                      |
| :----- | :------------------------ |
| `POST` | `/aws/static-roles/:name` |

### Parameters

- `name` `(string: <required>)` – Specifies the name of the role to create. This
is specified as part of the URL.

- `username` `(string: <required>)` – Specifies the username of the IAM user.

- `rotation_period` `(string/int: <required>)` – Specifies the amount of time
Vault should wait before rotating the password. The minimum is 1 minute. Can be

> speciied in either `24h` or `86400` format.

Reviewer 2: ```suggestion
specified in either `24h` or `86400` format.
```

> speciied in either `24h` or `86400` format.

Reviewer 2: Might help users to link to the [duration-format](https://developer.hashicorp.com/vault/docs/concepts/duration-format) similar to some other APIs that have duration fields.

Author: done!
